pip install pytetwild[all] fails from a clean conda env on Linux aarch64 (NVIDIA DGX Spark): CMake cannot find GMP

### Summary

`pip install pytetwild[all]` fails from a clean conda environment on Linux `aarch64` during the native build step for `pytetwild` / vendored `fTetWild`.

The build gets through Python dependency resolution and enters CMake configure, but then aborts because GMP is not found:

```text
-- Could NOT find GMP (missing: GMP_INCLUDE_DIRS GMP_LIBRARIES)
CMake Error at src/fTetWild/CMakeLists.txt:86 (message):
  Cannot find GMP
This was reproduced directly on the host machine, not in Docker.

Environment
Machine: NVIDIA DGX Spark
OS / kernel: Linux-6.14.0-1015-nvidia-aarch64-with-glibc2.39
Architecture: aarch64
Python: 3.14.3
Python distribution: Anaconda
Compiler: gcc/g++ 13.3.0
Install method: pip install "pytetwild[all]"
Environment manager: conda
Clean reproduction
From a fresh conda env:

bash

conda create -n pytetwild_repro python=3.14 -y
conda activate pytetwild_repro

python -m pip install --upgrade pip setuptools wheel
unset LD_LIBRARY_PATH

python -m pip install --no-cache-dir "pytetwild[all]"
Observed behavior
The wheel build fails during CMake configure for the vendored native dependency.

Relevant output:

text

*** scikit-build-core 0.12.2 using CMake 3.28.3 (wheel)
*** Configuring CMake...
loading initial cache file build/CMakeInit.txt
-- The CXX compiler identification is GNU 13.3.0
-- Found Python: /home/lammspark01/miniconda3/envs/pytetwild_repro/bin/python (found suitable version "3.14.3", minimum required is "3.10")
-- Found OpenMP: TRUE (found version "4.5")
...
-- Could NOT find GMP (missing: GMP_INCLUDE_DIRS GMP_LIBRARIES)
-- GMP libs: GMP_LIBRARIES-NOTFOUND GMP_INCLUDE_DIRS-NOTFOUND
CMake Error at src/fTetWild/CMakeLists.txt:86 (message):
  Cannot find GMP

-- Configuring incomplete, errors occurred!

*** CMake configuration failed
ERROR: Failed building wheel for pytetwild
Failed to build pytetwild
error: failed-wheel-build-for-install
Expected behavior
One of the following:

pip install pytetwild[all] succeeds on Linux aarch64, or
the installation documentation clearly states that GMP must be installed separately on this platform before invoking pip, or
the build failure message is surfaced more explicitly from the Python package side.
Additional environment info
text

python = 3.14.3 | packaged by Anaconda, Inc. | (main, Feb 24 2026, 22:54:32) [GCC 14.3.0]
platform = Linux-6.14.0-1015-nvidia-aarch64-with-glibc2.39
machine = aarch64

gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
g++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Notes
There is also a CMake developer warning during configure:

text

The package name passed to `find_package_handle_standard_args` (GMP) does
not match the name of the calling package (GMPfTetWild).
That warning is not the fatal issue in this run, but I am including it in case it is relevant to how GMP discovery is wired in the vendored build.

Question
Is Linux aarch64 expected to be supported for source installs today?

If yes, what is the recommended dependency installation path for GMP on this platform, and should that dependency be documented more explicitly for pip install pytetwild[all]?

--- a/.github/workflows/build-and-deploy.yml
+++ b/.github/workflows/build-and-deploy.yml
@@ -19,7 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-24.04, windows-2025, macos-latest]
+        os: [ubuntu-24.04, ubuntu-24.04-arm, windows-2025, macos-latest]
 
     steps:
       - uses: actions/checkout@v6
@@ -64,7 +64,7 @@ jobs:
       fail-fast: false
       matrix:
         python-version: ['3.13'] # , '3.14']  # skip 3.14 testing until vtk 9.6 is released
-        os: [ubuntu-24.04, windows-2025, macos-latest]  # must match build_wheels os
+        os: [ubuntu-24.04, ubuntu-24.04-arm, windows-2025, macos-latest]  # must match build_wheels os
     steps:
     - uses: actions/checkout@v6
     - uses: actions/setup-python@v6

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,9 +29,11 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ./src/pytetwild)
 # Set the path to the fTetWild project
 set(fTetWild_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src/fTetWild")
 
-# Enable aggressive optimization
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3")
-set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -O3")
+# Enable aggressive optimization. -fsigned-char keeps `char` signed on
+# aarch64 (where it defaults to unsigned); fTetWild stores -1 sentinels
+# in std::array<char, 4>, which is only valid with signed char.
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -fsigned-char")
+set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -O3 -fsigned-char")
 
 # Include the fTetWild project as a subdirectory
 add_subdirectory(${fTetWild_DIR} EXCLUDE_FROM_ALL)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -64,7 +64,9 @@ test-command = "pytest {project}/tests -v"
 manylinux-x86_64-image = "manylinux2014"
 manylinux-aarch64-image = "manylinux2014"
 before-all = "yum install -y gmp-devel"
-before-build = "pip install abi3audit cmake"
+# On aarch64, point fTetWild's vendored geogram at its aarch64
+# platform file; the default Linux64-gcc one injects x86-only -m64.
+before-build = "pip install abi3audit cmake && if [ $(uname -m) = aarch64 ]; then sed -i 's|VORPALINE_PLATFORM Linux64-gcc |VORPALINE_PLATFORM Linux64-gcc-aarch64 |' src/fTetWild/cmake/geogram.cmake; fi"
 repair-wheel-command = [
   "auditwheel repair -w {dest_dir} {wheel}",
   "bash tools/audit_wheel.sh {wheel}"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -62,6 +62,7 @@ test-command = "pytest {project}/tests -v"
 
 [tool.cibuildwheel.linux]
 manylinux-x86_64-image = "manylinux2014"
+manylinux-aarch64-image = "manylinux2014"
 before-all = "yum install -y gmp-devel"
 before-build = "pip install abi3audit cmake"
 repair-wheel-command = [

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -59,6 +59,11 @@ build = "cp310-* cp311-* cp312-*"  # 3.12+ are abi3 wheels
 skip = ["*musllinux*"]  # disable musl-based wheels
 test-extras = ["tests"]
 test-command = "pytest {project}/tests -v"
+# vtk (via pyvista) ships no manylinux2014 aarch64 wheel, so the
+# in-container test env cannot resolve test deps. The aarch64 wheel
+# still builds here; it is exercised by the separate test_abi3 job
+# on a native arm runner with a modern pip environment.
+test-skip = "*-manylinux_aarch64"
 
 [tool.cibuildwheel.linux]
 manylinux-x86_64-image = "manylinux2014"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -65,8 +65,8 @@ manylinux-x86_64-image = "manylinux2014"
 manylinux-aarch64-image = "manylinux2014"
 before-all = "yum install -y gmp-devel"
 # On aarch64, point fTetWild's vendored geogram at its aarch64
-# platform file; the default Linux64-gcc one injects x86-only -m64.
-before-build = "pip install abi3audit cmake && if [ $(uname -m) = aarch64 ]; then sed -i 's|VORPALINE_PLATFORM Linux64-gcc |VORPALINE_PLATFORM Linux64-gcc-aarch64 |' src/fTetWild/cmake/geogram.cmake; fi"
+# platform; the default Linux64-gcc one injects x86-only -m64.
+before-build = "pip install abi3audit cmake && if [ $(uname -m) = aarch64 ]; then sed -i 's|GEO_PLATFORM \"Linux64-gcc\"|GEO_PLATFORM \"Linux64-gcc-aarch64\"|' src/fTetWild/cmake/FloatTetwildDependencies.cmake; fi"
 repair-wheel-command = [
   "auditwheel repair -w {dest_dir} {wheel}",
   "bash tools/audit_wheel.sh {wheel}"

